crates/sui/src/ptb/ptb_parser/argument.rs

```diff
@@ -0,0 +1,145 @@
+            Argument::U64(u) => write!(f, "{}u64", u),
+            Argument::U128(u) => write!(f, "{}u128", u),
+            Argument::U256(u) => write!(f, "{}u256", u),
+            Argument::Gas => write!(f, "gas"),
+            Argument::Identifier(i) => write!(f, "{}", i),
+            Argument::VariableAccess(head, accesses) => {
+                write!(f, "{}", head)?;
+                for access in accesses {
+                    write!(f, ".{}", access)?;
+                }
+                Ok(())
+            }
+            Argument::Address(a) => write!(f, "@{}", a),
+            Argument::String(s) => write!(f, "\"{}\"", s),
+            Argument::Vector(v) => {
+                write!(f, "vector[")?;
+                for (i, arg) in v.iter().enumerate() {
+                    write!(f, "{}", arg)?;
+                    if i != v.len() - 1 {
+                        write!(f, ", ")?;
+                    }
+                }
+                write!(f, "]")
+            }
+            Argument::Array(a) => {
+                write!(f, "[")?;
+                for (i, arg) in a.iter().enumerate() {
+                    write!(f, "{}", arg)?;
+                    if i != a.len() - 1 {
+                        write!(f, ", ")?;
+                    }
+                }
+                write!(f, "]")
+            }
+            Argument::Option(o) => match o {
+                Some(v) => write!(f, "some({})", v),
+                None => write!(f, "none"),
+            },
+            Argument::ModuleAccess {
+                address,
+                module_name,
+                function_name,
+            } => {
+                write!(f, "{}::{}::{}", address, module_name, function_name)
+            }
+            Argument::TyArgs(ts) => {
+                write!(f, "<")?;
+                for (i, t) in ts.iter().enumerate() {
+                    write!(f, "{}", t.clone().into_type_tag(&resolve_address).unwrap())?;
+                    if i != ts.len() - 1 {
+                        write!(f, ", ")?;
+                    }
+                }
+                write!(f, ">")
+            }
+        }
+    }
+}
+
+impl Argument {
+    pub fn into_move_value_opt(&self) -> Result<MoveValue> {
+        Ok(match self {
+            Argument::Bool(b) => MoveValue::Bool(*b),
+            Argument::U8(u) => MoveValue::U8(*u),
+            Argument::U16(u) => MoveValue::U16(*u),
+            Argument::U32(u) => MoveValue::U32(*u),
+            Argument::U64(u) => MoveValue::U64(*u),
+            Argument::U128(u) => MoveValue::U128(*u),
+            Argument::U256(u) => MoveValue::U256(*u),
+            Argument::Address(a) => MoveValue::Address(a.into_inner()),
+            Argument::Vector(vs) => MoveValue::Vector(
+                vs.iter()
+                    .map(|v| v.into_move_value_opt())
+                    .collect::<Result<Vec<_>>>()
+                    .with_context(|| format!(
```

Comment:
I am on the edge with using `with_context`. I often don't like how it repeats the error info in the error message and then in the context bottom part. I'd need to try it out and see how things look like. Your thoughts here?

Reply:
Personally, I think `with_context` is fine -- you just need to be careful in how you build up the stack of error messages so that you avoid the repetition that you mention (which is easy to do if you're not careful!)